src/duk_debugger.c

```diff
@@ -1448,6 +1448,49 @@ DUK_LOCAL void duk__debug_handle_detach(duk_hthread *thr, duk_heap *heap) {
 	DUK__SET_CONN_BROKEN(thr, 0);  /* not an error */
 }
 
+DUK_LOCAL void duk__debug_handle_apprequest(duk_hthread *thr, duk_heap *heap) {
+	DUK_D(DUK_DPRINT("debug command AppRequest"));
+
+	if (heap->dbg_request_cb != NULL) {
+		duk_context *ctx = (duk_context *) thr;
+		duk_idx_t nrets;
+		duk_idx_t nvalues = 0;
+		duk_idx_t old_top;
+		duk_idx_t top, idx;
+
+		/* Read tvals from the message and push them onto the valstack,
+		 * then call the request callback to process the request.
+		 */
+		old_top = duk_get_top(ctx);  /* save stack top */
+		while (duk_debug_peek_byte(thr) != DUK_DBG_MARKER_EOM) {
```

Comment:
To support very large number of values there should be a value stack extend check here. Because we don't support errors in the debugger code, it should be something like:

```
if (!duk_check_stack(ctx, 1)) {
    DUK_D(DUK_DPRINT("failed to allocate space for notify dvalue(s)"));
    DUK__SET_CONN_BROKEN(thr, 0);
    return;
}
```

The need for this should be exercisable by sending an apprequest with say 256 arguments. I tried this right now and this happens without the check:

    [D] duk_debugger.c:1452 (duk__debug_handle_apprequest): debug command AppRequest
    [D] duk_js_executor.c:1149 (duk__handle_longjmp): -> throw propagated up to entry level, rethrow and exit bytecode executor
    Error: attempt to push beyond currently allocated stack
        duk_api_stack.c:3103
        global dist-files/mandel.js:8 preventsyield
    [...]

That error propagated out of the debugger which is bad right now.

With the added check it works; a detach will happen on out-of-memory.